### PR TITLE
Add endpoint view to the Digma Metrics report

### LINKED ISSUE #1046: Heatmap - Drilldown to entry points from service
See mockup 
https://www.figma.com/proto/MxA95ZHK5NqGX4ptFQljRJ/%E2%AD%90-Design-System-3.0?page-id=13456%3A263745&node-id=13456-266377&viewport=253%2C-6467%2C1.02&t=cqX5WC7LZus7PPy9-1&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=13456%3A266377

Clicking on a service name should show a heatmap of all entry points as seen on the mockup.
In the transition we should keep following values from the previous page we came from:
1. Environment
2. Period
3. Changes/Baseline value
4. Tab type (Table/Heatmap)

If there are no entry points for that service under the selected environment then we need to have 'no entry points recorded for this environment'. @ionuiux could you please add?

Environment selector:
The user can choose to select a different environment from this page in the filter control. The list of selectable environments should be all environments the service was recorded under.

Filters:
For this story, include only the following components in the filters header:
* Environment selector
* Period (in the changes view)
* Toggle between heatmap/table visualization

### LINKED ISSUE #1049: Heatmap - Clicking on endpoint issues should open issues view for that endpoint
When the user clicks on the endpoint issues link in the endpoint drilldown view for a service, it should change the scope in the main panel to that endpoint and open the issues tab. The selected environment should also be the environment currently selected in the heatmap document.

Data and Endpoints filters are  not part of this issue

In the table view you can see the issues link when you hover on the value of the issues count and impact value:

### LINKED ISSUE #1058: Endpints filter for endpoint heatmap
The component should show all endpoints under the service. Similar to the services filter in the services heatmap, only the selected endpoints data will be displayed.

### LINKED ISSUE #1074: services issues - dashboard issues
issues
- [x] scaling issues when moving the cursor check the attached video 
Fixed in #1070

[Private Zenhub Video](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBLzcrQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--ac7e0bf7381a9b88f7d53890527361b46e8eaff7/Screen%20Recording%202024-09-25%20at%204.48.44%20PM.mov)

- [x] The user can unselect an environment, it seems wrong.

[Private Zenhub Video](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeHIvQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--05329b27885a999a3eab3700cf8042642e98e958/Screen%20Recording%202024-09-25%20at%205.08.43%20PM.mov)

### PR DESCRIPTION
**Pull request by kshmidt-digma**
Closes #1046 
Closes #1058 
Closes #1049 
Closes #1074

### CODE REVIEW

File: src/components/Dashboard/NewReport/Chart/Chart.stories.tsx [hunk 1]
```diff
@@ -20,6 +21,12 @@ type Story = StoryObj<typeof meta>;
 // More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
 export const Default: Story = {
   args: {
-    data: mockedReport.reports
+    scoreCriterion: "criticality",
+    timeMode: "baseline",
+    viewLevel: "services",
+    onTitleClick: () => {
```
Comment (opoliarush) [inline src/components/Dashboard/NewReport/Chart/Chart.stories.tsx:27]: fn()

File: src/components/Dashboard/NewReport/ReportHeader/index.tsx [hunk 2]
```diff
@@ -102,13 +152,18 @@ export const ReportHeader = ({
...
+    periodInDays,
+    selectedEndpoints
   });
 
   useEffect(() => {
     getData();
   }, []);
 
+  useEffect(() => {
+    getEndpointsData();
```
Comment (opoliarush) [inline src/components/Dashboard/NewReport/ReportHeader/index.tsx:164]: no need to trigger get endpoint data for services view

File: src/components/Dashboard/NewReport/ReportHeader/index.tsx [hunk 3]
```diff
@@ -193,10 +252,41 @@ export const ReportHeader = ({
     setTimeMode(newMode);
   };
 
+  const environmentsToSelect = service
+    ? serviceEnvironmentsData?.environments ?? []
+    : environments ?? [];
```
Comment (opoliarush) [inline src/components/Dashboard/NewReport/ReportHeader/index.tsx:257]:
```suggestion
  const environmentsToSelect = (service
    ? serviceEnvironmentsData?.environments 
    : environments ) ??  [];
```

File: src/components/Dashboard/NewReport/index.tsx [hunk 4]
```diff
@@ -46,49 +80,147 @@ export const NewReport = () => {
...
+      goToEndpointIssues({
+        spanCodeObjectId: value,
+        service: selectedService,
+        environmentId: query.environmentId
+      });
+    }
+  };
+
+  const handleGoBack = () => {
+    setSelectedService(undefined);
```
Comment (opoliarush) [inline src/components/Dashboard/NewReport/index.tsx:144]: can we use null instead of undeifned?

File: src/components/Main/types.ts [hunk 5]
```diff
@@ -56,8 +56,10 @@ export enum SCOPE_CHANGE_EVENTS {
   RECENT_ACTIVITY_SPAN_LINK_CLICKED = "RECENT_ACTIVITY_SPAN_LINK_CLICKED",
   IDE_CODE_LENS_CLICKED = "IDE/CODE_LENS_CLICKED",
   IDE_NOTIFICATION_LINK_CLICKED = "IDE/NOTIFICATION_LINK_CLICKED",
+  IDE_CUSTOM_PROTOCOL_LINK_CLICKED = "IDE/CUSTOM_PROTOCOL_LINK_CLICKED",
```
Comment (opoliarush) [inline src/components/Main/types.ts:59]: Can we extract all IDE events on level upper?

Comment (kshmidt-digma) [reply src/components/Main/types.ts:59]: SCOPE_CHANGE events are currently related to `Main` app only

**Review by opoliarush — APPROVED** [review]

**Pull request merged**